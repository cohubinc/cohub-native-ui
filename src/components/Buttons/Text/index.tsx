import React, { PureComponent, CSSProperties } from 'react';
import Blank from '../Blank';
import Color from '../../../definitions/enums/Color';
import Typography from '../../Typography';
import { Props as BlankButtonProps } from '../../Buttons/Blank';

interface IProps {
  color?: Color;
  fontSize?: number | string;
  textStyle?: CSSProperties;
  block?: boolean;
}

type TProps = IProps & Omit<BlankButtonProps, 'color'>;

export default class Text extends PureComponent<TProps> {
  static defaultProps: Partial<IProps> = {
    color: Color.iconGrey,
    fontSize: '14px',
  };

  render() {
    const {
      color,
      fontSize,
      children,
      textStyle,
      block,
      style,
      className = '',
      ...rest
    } = this.props;

    return (
      <Blank
        {...rest}
        className={`${className} p-05`}
        style={{ display: block ? 'block' : undefined, ...style }}
      >
        <Typography
          uppercase
          color={color}
          weight={500}
          style={{
            fontSize,
            ...textStyle,
          }}
          kerning={0.07}
        >
          {children}
        </Typography>
      </Blank>
    );
  }
}
