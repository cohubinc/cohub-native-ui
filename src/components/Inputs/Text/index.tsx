import React from "react";
import { FieldRenderProps } from "react-final-form";

import TInputElementProps from "../definitions/TInputElementProps";
import Base from "../Base";

type FieldProps = FieldRenderProps<string, HTMLInputElement>;

interface IProps {
  label?: string;
  appearance?: "contrast" | "inverted";
  msgPosition?: {
    bottom: number;
  };
  "data-qa"?: string;
  input?: Partial<FieldProps["input"]>;
  meta?: FieldProps["meta"];
}

type TProps = IProps & Omit<TInputElementProps, "onChange" | "value">;

export default function Text(props: TProps) {
  const {
    input,
    meta,
    style,
    className,
    msgPosition = {},
    "data-qa": dataQa,
    ...restProps
  } = props;

  const { touched, error } = meta || ({} as any);

  const showError = !!(touched && error);

  return (
    <div {...{ className, style: { width: "100%", ...style } }}>
      <Base {...input} {...restProps} error={showError} data-qa={dataQa} />
    </div>
  );
}
