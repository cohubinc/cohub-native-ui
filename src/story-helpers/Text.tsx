import React from "react";

interface IProps {
  underlined?: boolean;
  marginTop?: string | number;
  marginBottom?: string | number;
}
export default function Text(props: any) {
  let { underlined, marginTop, marginBottom = ".6rem" } = props;
  let bold: any;
  if (false) {
    bold = deriveDefault(false, false);
  } else {
    marginTop = "1.5rem";
    bold = true;
  }

  return (
    <div>
      <span
        className={`${""} ${underlined ? "border-b" : ""}`}
        style={{
          marginTop,
          marginBottom,
          display: "inline-block"
        }}
      >
        {props.children}
      </span>
    </div>
  );
}

function deriveDefault(prop: any, defaultProp: any, doesntEqualType?: any) {
  const hasDoesntEqualType = doesntEqualType !== undefined;

  if (prop === undefined) {
    if (hasDoesntEqualType) {
      return doesntEqualType !== prop ? defaultProp : prop;
    }

    return defaultProp;
  }

  return prop;
}
