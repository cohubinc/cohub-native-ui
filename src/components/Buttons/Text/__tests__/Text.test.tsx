import React from "react";
import { mount } from "enzyme";

import Text from "../index";

describe("Text", () => {
  it("renders without crashing", () => {
    const wrapper = mount(<Text>push me</Text>);
    expect(wrapper).toMatchSnapshot();
  });
});
