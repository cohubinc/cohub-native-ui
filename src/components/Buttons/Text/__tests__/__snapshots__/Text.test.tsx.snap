// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders without crashing 1`] = `
<Text
  color="hsl(212, 14%, 35%)"
  fontSize="14px"
>
  <Blank
    className=" p-05"
    style={
      Object {
        "display": undefined,
      }
    }
  >
    <button
      className="ButtonBlank  p-05"
      style={
        Object {
          "display": undefined,
        }
      }
    >
      <Typography
        style={Object {}}
      >
        <Component
          style={Object {}}
        >
          <span
            className=""
            data-qa="text"
            style={
              Object {
                "color": "hsl(0, 2%, 22%)",
                "fontFamily": "Inter",
                "fontSize": "14px",
                "fontWeight": 400,
                "textAlign": "initial",
                "textTransform": "initial",
              }
            }
          >
            <Typography
              color="hsl(212, 14%, 35%)"
              kerning={0.07}
              style={
                Object {
                  "fontSize": "14px",
                }
              }
              uppercase={true}
              weight={500}
            >
              <Component
                color="hsl(212, 14%, 35%)"
                kerning={0.07}
                style={
                  Object {
                    "fontSize": "14px",
                  }
                }
                uppercase={true}
                weight={500}
              >
                <span
                  className=""
                  data-qa="text"
                  style={
                    Object {
                      "color": "hsl(212, 14%, 35%)",
                      "fontFamily": "Inter",
                      "fontSize": "14px",
                      "fontWeight": 500,
                      "letterSpacing": "0.07rem",
                      "textAlign": "initial",
                      "textTransform": "uppercase",
                    }
                  }
                >
                  push me
                </span>
              </Component>
            </Typography>
          </span>
        </Component>
      </Typography>
    </button>
  </Blank>
</Text>
`;
